### PR TITLE
support of ctrl and shift key & simple formatter

### PR DESCRIPTION
**Pull request by Jiramew**
1、support of ctrl and shift key which could +10, +0.1 times of step          
2、simple formatter, with the function eg:          
```
format (num) {
  return "$ "+ num
}
```
3、bug fix, the value still changes when pressing up or down key even with the readonly status

Thanks, Sincerely
Jiramew

### CODE REVIEW

**Comment by afc163** [comment]
I will check this later in this week, could you add some test cases?

**Comment by Jiramew** [comment]
Add some test cases for the bug fixed and the formatter, still don't know how to test/simulate combination keys (eg. <kbd>Shift + ⬆</kbd>) properly though.

Thanks,
Jiramew

File: examples/simple.js [hunk 1]
```diff
@@ -26,17 +26,23 @@ const Component = React.createClass({
...
-          min={-8}
-          max={10}
+          min={-8000}
+          max={10000}
           value={this.state.value}
           style={{ width: 100 }}
           readOnly={this.state.readOnly}
           onChange={this.onChange}
           disabled={this.state.disabled}
+          autoFocus={false}
```
Comment (afc163) [inline examples/simple.js]: It is better to create another demo than modify this one.

File: src/index.js [hunk 2]
```diff
@@ -54,9 +55,21 @@ const InputNumber = React.createClass({
 
   onKeyDown(e, ...args) {
     if (e.keyCode === 38) {
-      this.up(e);
+      let ratio = 1;
+      if (e.metaKey || e.ctrlKey) {
+        ratio = 0.1;
+      } else if (e.shiftKey) {
+        ratio = 10;
+      }
```
Comment (afc163) [inline src/index.js]: You can put those ratio code to a `this.getRatio` function.

**Comment by afc163** [comment]
Could you add some API document and instructions.

File: tests/index.js [hunk 3]
```diff
@@ -591,4 +624,131 @@ describe('inputNumber', () => {
...
+
+  describe('formatter', () => {
+    it('formatter on default', () => {
+      const Demo = React.createClass({
+        render() {
+          return (<InputNum
+            ref="inputNum"
+            step={1}
+            value={5}
+            formatter={(num) => {return `$ ${num}`;}}
```
Comment (afc163) [inline tests/index.js]:
```diff
- formatter={(num) => {return `$ ${num}`;}}
+ formatter={num => `$ ${num}`}
```

File: README.md [hunk 4]
```diff
@@ -199,6 +199,12 @@ react-native run-ios
           <td></td>
           <td>custom the down step element</td>
         </tr>
+        <tr>
+          <td>formatter</td>
+          <td>Function</td>
+          <td></td>
+          <td>Specifies the format of the value presented</td>
+        </tr>
```
Comment (afc163) [inline README.md]: Also document short keys?

Comment (afc163) [inline README.md]: like: https://github.com/vlad-ignatov/react-numeric-input#keyboard-navigation

**Comment by afc163** [comment]
There is a tiny problem.

1. Pressing <kbd>cmd</kbd> + <kbd>↑</kbd>, number increase as expected.
2. Release <kbd>↑</kbd>, then number is still increasing.

**Comment by Jiramew** [comment]
Hi～I'll check on that~ Sorry for my late reply~

**Pull request merged**